sdk/containerregistry/azure-containerregistry/azure/containerregistry/_exchange_client.py

```diff
@@ -0,0 +1,127 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import re
+from typing import TYPE_CHECKING
+
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+from ._generated import ContainerRegistry
+from ._user_agent import USER_AGENT
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+    from azure.core.pipeline import PipelineRequest, PipelineResponse
+    from typing import Dict, Any, List
+
+
+class ExchangeClientAuthenticationPolicy(SansIOHTTPPolicy):
```

Comment:
So if I have the challenge header from a prior request - no further authentication is required?

Reply:
Not necessarily, the scopes are different for virtually every operation so the challenge still needs to be completed. In the future, caching will be added to reduce the number of extra calls.